backend/src/main/java/com/example/cms/user/UserService.java

```diff
@@ -155,10 +183,25 @@ public UserDtoDetailed updateUser(Long id, UserDtoFormUpdate form) {
                 user.setEnrolledUniversities(this.validateUniversities(form.getEnrolledUniversities()));
             }
         }
+    }
+
+    @Secured("ROLE_USER")
+    public UserDtoDetailed updateUser(Long id, UserDtoFormUpdate form) {
+        User user = validateUserAndForm(id, form);
+        boolean mainDataNotChanged = mainDataNotChanged(user, form);
+        String oldEmail = user.getEmail();
+        updateUserDetails(user, form);
+        handleUpdateAccountStatus(user, form);
 
         if (securityService.hasHigherRoleThan(Role.MODERATOR) && !form.getPassword().isEmpty()) {
             validatePassword(form.getPassword());
             user.setPassword(passwordEncoder.encode(form.getPassword()));
+            emailService.sendEditUserAccountMail(
+                    oldEmail, user, "administrator@reunice.pl", "admin", form.getPassword());
+        } else {
+            if (!mainDataNotChanged) {
+                emailService.sendEditUserAccountMail(oldEmail, user, "administrator@reunice.pl", "admin");
```

Comment:
Wysyłanie chyba lepiej zrobić po wywołaniu `repository.save()` bo jeśli jest jakiś błąd walidacji (np. za długa nazwa użytkownika), to dopiero `.save()` to zwróci. W takim wypadku może pójść mail o zmianie hasła, a sama edycja się na przykład nie powiedzie.